content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,174 @@
+
+### Info
+
+Use the `info` state to highlight messages that help users complete a task or that provide additional context.
+
+Examples:
+- Single sign on or other account related notifications
+- General information about a feature or setting
+
+<img
+  width="960"
+  alt="A single sign on info banner on the GitHub homepage"
+  src="https://github.com/primer/design/assets/18661030/1055049e-6db8-41d5-8cf8-e375bd0e8dea"
+/>
+
+### Warning
+
+Use the `warning` state to inform users of a potential issue, an important upcoming change that may impact their experience or a potentially unexpected consequence of an action.
+
+Examples:
+- An account setting will become required soon
+- A [degraded experience](./degraded-experiences) due to connectivity or GitHub availability
+- A non-destructive action with side-effects like costs or access changes is about to take place
+
+<img
+  width="960"
+  alt="A global warning banner above GitHub's global navigation bar inside a browser window."
+  src="https://github.com/primer/design/assets/18661030/468b73b1-de8b-45e0-b8bb-325516df942e"
+/>
+
+### Success
+
+Use the `success` state to notify users that the results of an action were successful if it is not apparent from other parts of the UI. **Use success messaging sparingly** and rely more on interaction context.
+
+Examples:
+- Saving an account setting if there's no page reload or redirect
+- A background process has successfully completed
+
+### Critical
+
+Use the `critical` state to notify users that an action was unsuccessful, to display form errors, or to inform that an action is destructive.
+
+Examples:
+- Form field is invalid upon submitting
+- A delete or transfer button
+
+<img
+  width="960"
+  alt="A critical error message on an input field within GitHub Issues."
+  src="https://github.com/primer/design/assets/18661030/5da2e479-6bc7-46dc-9179-46002412b9f5"
+/>
+
+### Neutral
+
+Reserve the `neutral` state for **inline** system related degraded experiences, or for surfacing account related permission messages.
+
+Examples:
+- A row of a data table could not be loaded
+- An item in a list could not be loaded
+- User does not have permissions to perform an action
+
+See [degraded experiences](./degraded-experiences) for more information.
+
+### Upsell
+
+Use the `upsell` state to highlight features that could be unlocked if the user upgrades their account.
+
+Examples:
+- Upgrade account to access a feature
+
+<img
+  width="960"
+  alt="An upsell banner inside the rulesets setting page on GitHub explaing that this feature can be used if the user upgrades to an Enterprise account."
+  src="https://github.com/primer/design/assets/18661030/924884b8-4b85-4967-898d-e0c2fcfaf785"
+/>
+
+## Placement
+
+Message proximity contributes to the prominence of a message. Whenever possible, place messages near the action they are related to.
```

Comment:
@langermank would this placement section be a good place for me to put the more detail explanation of when to use banner vs inline message, along with the flowchart? Or should that be its own section?